File: cs/Minsk/CodeAnalysis/Syntax/UnaryExpressionSyntax.cs
﻿namespace Minsk.CodeAnalysis.Syntax;

public sealed class UnaryExpressionSyntax : ExpressionSyntax
{
    public UnaryExpressionSyntax(SyntaxToken operatorToken, ExpressionSyntax operand)
    {
        OperatorToken = operatorToken;
        Operand = operand;
    }

    public SyntaxToken OperatorToken { get; }
    public ExpressionSyntax Operand { get; }

    public override SyntaxKind Kind => SyntaxKind.UnaryExpression;
    public override IEnumerable<SyntaxNode> Children => new SyntaxNode[] { OperatorToken, Operand };
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/UnaryExpressionSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class UnaryExpressionSyntax(val operatorToken: SyntaxToken, val operand: ExpressionSyntax) : ExpressionSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.UnaryExpression
    override val children: List<SyntaxNode>
        get() = listOf(operatorToken, operand)
}
